src/yunohost/backup.py

```diff
@@ -467,14 +476,22 @@ def _is_hook_in_backup(h):
 
                 # Execute app restore script
                 subprocess.call(['install', '-Dm555', app_script, tmp_script])
-                hook_exec(tmp_script, args=[tmp_app_bkp_dir, app_id],
-                          raise_on_error=True, chdir=tmp_app_bkp_dir)
+
+                # Prepare env. var. to pass to script
+                env_dict = {}
+                app_id, app_instance_nb = _parse_app_instance_name(app_instance_name)
+                env_dict["YNH_APP_ID"] = app_id
+                env_dict["YNH_APP_INSTANCE_NAME"] = app_instance_name
+                env_dict["YNH_APP_INSTANCE_NUMBER"] = str(app_instance_nb)
+
+                hook_exec(tmp_script, args=[tmp_app_dir + '/backup', app_instance_name],
```

Comment:
`args` argument value has changed, it should be `args=[tmp_app_bkp_dir, app_instance_name],`.